src/Npgsql/NpgsqlReadBuffer.Stream.cs

```diff
@@ -111,75 +111,94 @@ public override Task FlushAsync(CancellationToken cancellationToken)
                 => throw new NotSupportedException();
 
             public override int Read(byte[] buffer, int offset, int count)
-                => Read(buffer, offset, count, CancellationToken.None, false).GetAwaiter().GetResult();
+                => Read(new Span<byte>(buffer, offset, count));
 
             public override Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
             {
                 using (NoSynchronizationContextScope.Enter())
-                    return Read(buffer, offset, count, cancellationToken, true).AsTask();
+                    return ReadAsync(new Memory<byte>(buffer, offset, count), cancellationToken).AsTask();
             }
 
-            ValueTask<int> Read(byte[] buffer, int offset, int count, CancellationToken cancellationToken, bool async)
+#if !NET461 && !NETSTANDARD2_0
+            public override int Read(Span<byte> span)
+#else
+            public int Read(Span<byte> span)
```

Comment:
This is wrong - we may need to call ReadLong etc.

Reply:
I do not understand what is wrong here. Can you please detail?

Reply:
As it's written this doesn't ever perform any I/O, which it needs to do for sequential readers... This is because the wrong overload of ReadBytes is being called below. The ReadBytes you're calling here (and in the async version) only copies existing buffered bytes from NpgsqlReadBuffer into the provided Span; if you look at the original code before your PR, it calls a ReadBytes overload that performs I/O if necessary (which is why it accepts the async flag).

It's unfortunate we have functions named the same doing such different actions - at some point we should generally clean up the buffers.

Reply:
OK. I understand the issue now. 💡
I have renamed those methods according to the `Read(Span)`/`ReadAsync(Memory)` usual convention.